src/ssh_config/parser.rs

```diff
@@ -84,10 +84,16 @@ impl Parser {
                     }
                 }
                 EntryType::Host => {
+                    if let Some(host) = current_host.take() {
+                        hosts.push(host);
+                    }
                     let patterns = parse_patterns(&entry.1);
-                    hosts.push(Host::new(patterns));
-                    is_in_host_block = true;
-
+                    if patterns.contains(&"*".to_string()) {
+                        global_host = Host::new(patterns.clone());
+                        hosts.push(global_host.clone());
```

Comment:
This would add an host with patterns but empty entries. I'm not sure if it's needed 🤔 `global_host` by itself may not be needed as we have `current_host` now.